main.tf

```diff
@@ -54,9 +66,17 @@ EOT
 
 # EKS Provider
 provider "kubernetes" {
+  # This defers provider initialization until the cluster is ready
   host                   = data.aws_eks_cluster.cluster.endpoint
   cluster_ca_certificate = base64decode(local.kubeconfig_ca_cert)
   token                  = data.aws_eks_cluster_auth.cluster.token
+
+  # This keeps the token up-to-date during subsequent applies, even if they run longer than the token TTL.
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--cluster-name", module.eks.cluster_id]
+    command     = "aws"
```

Comment:
Guessing we already have eks on the running system? This now creates another dependency on an external application. Are you sure the info did not update during each apply? I never had an issue where an new apply could not access a provider based kubeconfig. Each time the apply was run it re-upped the token.

Reply:
It's an existing dependency. The syntax makes it appear to be "eks" but it is the eks sub-command of the aws cli command which is already a dependency. I added this to try to extend the 15 minute TTL of the token based on advice that others had good results with it. I am inclined to remove it since I don't think it appeared to refresh the token within an apply run as others claimed it could. As you say, without it each apply generates a new token good for another 15 mins.

Reply:
Good to know on the dependency. If it's not useful, I would say remove it as it will cause other headaches and confusion down the road.

Reply:
removed it